src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -215,6 +212,9 @@ public override Task AnalyzeProjectAsync(Project project, bool semanticsChanged,
                     projectInfo = await SymbolTreeInfo.GetInfoForSourceAssemblyAsync(
                         project, checksum, cancellationToken).ConfigureAwait(false);
 
+                    Contract.ThrowIfNull(projectInfo);
+                    Contract.ThrowIfTrue(projectInfo.Checksum != checksum, "If we computed a SymbolTreeInfo, then its checksum much match our checksum.");
```

Comment:
I'd prefer to not combine.  That way we know which of the checks failed.  If it's a single contract call, then it might be htat we got null back, or it might be that the checksums don't match.